djangocms_navigation/test_utils/factories.py

```diff
@@ -0,0 +1,142 @@
+import string
+
+from django.contrib.auth.models import User
+from django.contrib.contenttypes.models import ContentType
+from django.contrib.sites.models import Site
+
+from cms.models import Page, PageContent, TreeNode
+
+from djangocms_versioning.models import Version
+
+import factory
+from factory.fuzzy import FuzzyText, FuzzyChoice, FuzzyInteger
+
+from ..models import Menu, MenuContent, MenuItem
+
+
+class UserFactory(factory.django.DjangoModelFactory):
+    username = FuzzyText(length=12)
+    first_name = factory.Faker('first_name')
+    last_name = factory.Faker('last_name')
+    email = factory.LazyAttribute(
+        lambda u: "%s.%s@example.com" % (u.first_name.lower(), u.last_name.lower()))
+
+    class Meta:
+        model = User
+
+    @classmethod
+    def _create(cls, model_class, *args, **kwargs):
+        """Override the default ``_create`` with our custom call."""
+        manager = cls._get_manager(model_class)
+        # The default would use ``manager.create(*args, **kwargs)``
+        return manager.create_user(*args, **kwargs)
+
+
+class AbstractVersionFactory(factory.DjangoModelFactory):
+    object_id = factory.SelfAttribute('content.id')
+    content_type = factory.LazyAttribute(
+        lambda o: ContentType.objects.get_for_model(o.content))
+    created_by = factory.SubFactory(UserFactory)
+
+    class Meta:
+        exclude = ['content']
+        abstract = True
+
+
+class TreeNodeFactory(factory.django.DjangoModelFactory):
+    site = factory.fuzzy.FuzzyChoice(Site.objects.all())
+    depth = 0
+    # NOTE: Generating path this way is probably not a good way of
+    # doing it, but seems to work for our present tests which only
+    # really need a tree node to exist and not throw unique constraint
+    # errors on this field. If the data in this model starts mattering
+    # in our tests then something more will need to be done here.
+    path = FuzzyText(length=8, chars=string.digits)
```

Comment:
A bit off context comment here but we might need to think about how would we deal to create a version when path has alphabet like this `000E`?

Reply:
Yeah... True. The way I see it, I'll figure that out once the problem arises in our tests ;) So far this works, but yeah - we'll probably need a better way of generating the path eventually.